### PR TITLE
Fixed exchange rate calculation

### PR DESCRIPTION
**Pull request by grandizzy**
Added test for removing quote token with interest
Fixed failing tests

### CODE REVIEW

File: src/ERC20Pool.sol [hunk 1]
```diff
@@ -220,10 +220,14 @@ contract ERC20Pool is IPool, Clone {
             revert InvalidPrice();
         }
 
-        accumulatePoolInterest();
-
         // remove from bucket with RAD precision
         _amount = Maths.wadToRad(_amount);
+        if (_amount > totalQuoteToken) {
```
Comment (EdNoepel) [inline src/ERC20Pool.sol]: I don't see benefit forcing lenders to spend gas for a pool-level check like this.  It would only save gas in a corner case where the pool is approaching 100% utilization, because the caller won't need to test whether reallocation is possible.  `Bucket.removeQuoteToken` should handle this.

File: src/_test/ERC20PoolCollateral.t.sol [hunk 2]
```diff
@@ -264,7 +264,7 @@ contract ERC20PoolCollateralTest is DSTestPlus {
         vm.expectRevert(
             abi.encodeWithSelector(
                 Buckets.InsufficientLpBalance.selector,
-                1_000 * 1e27
+                999.999999999999998 * 1e27
```
Comment (EdNoepel) [inline src/_test/ERC20PoolCollateral.t.sol:267]: `bucket.claimCollateral` reverts a `ClaimExceedsCollateral` with the whole `bucket.collateral`.  Shouldn't this number be `0.374913050298415729988389873 * 1e27`?

File: src/libraries/Buckets.sol [hunk 3]
```diff
@@ -430,7 +430,7 @@ library Buckets {
             bucket.collateral *
             bucket.price;
         if (size != 0 && bucket.lpOutstanding != 0) {
-            size / bucket.lpOutstanding;
+            return size / bucket.lpOutstanding;
```
Comment (EdNoepel) [inline src/libraries/Buckets.sol:433]: 😮 How did this ever work?

File: src/_test/ERC20PoolCollateral.t.sol (hunk 2 above)
Comment (grandizzy) [reply src/_test/ERC20PoolCollateral.t.sol:267]: as discussed, that's the case for not enough redemption tokens

File: src/libraries/Buckets.sol (hunk 3 above)
Comment (grandizzy) [reply src/libraries/Buckets.sol:433]: was introduced with new precisions :see_no_evil:

File: src/ERC20Pool.sol (hunk 1 above)
Comment (grandizzy) [reply src/ERC20Pool.sol]: that's reverted now

**Review by EdNoepel — APPROVED** [review]
LGTM
